tests/ecosystem/upgrade/test_upgrade.py

```diff
@@ -1,66 +1,36 @@
 import logging
 
+import pytest
+
 from ocs_ci.framework.testlib import ocs_upgrade
-from ocs_ci.framework import config
-from ocs_ci.ocs.cluster import CephCluster, CephHealthMonitor
-from ocs_ci.ocs.resources.storage_cluster import (
-    ocs_install_verification
-)
-from ocs_ci.utility.utils import TimeoutSampler
-from ocs_ci.ocs.ocs_upgrade import (
-    verify_image_versions,
-    OCSUpgrade,
-)
-from ocs_ci.ocs.exceptions import TimeoutException
+from ocs_ci.ocs.ocs_upgrade import run_ocs_upgrade, UpgradeDisruptiveOps
+
 
 log = logging.getLogger(__name__)
 
+dist = UpgradeDisruptiveOps()
+
+
+@pytest.mark.parametrize(
+    argnames=["operation", "operation_args"], argvalues=[
+        pytest.param(
+            dist.worker_node_shutdown, True, marks=pytest.mark.polarion_id("OCS-1575")
+        ),
+        pytest.param(
+            dist.worker_node_shutdown, False, marks=pytest.mark.polarion_id("OCS-1579")
+        ),
+    ]
+)
+def test_worker_node_shutdown(nodes, operation, operation_args):
+    log.info("Starting disruptive test function: test_worker_node_shutdown")
+    run_ocs_upgrade(nodes, operation, operation_args)
+
 
 @ocs_upgrade
-def test_upgrade():
+def test_upgrade(nodes):
```

Comment:
Usage of nodes fixture here looks like overkill to me which adding a lot of complexity where you need just to pass it everywhere else.

Instead of passing it everywhere. Please use directly call of:
platform_nodes.PlatformNodesFactory().get_nodes_platform()
Just on the place where you really need those nodes which can be for example your operation?